pkg/backup/backup_compaction.go

```diff
@@ -45,8 +48,81 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 	"github.com/cockroachdb/errors"
 	"github.com/gogo/protobuf/types"
+	"github.com/lib/pq"
 )
 
+var (
+	backupCompactionThreshold = settings.RegisterIntSetting(
+		settings.ApplicationLevel,
+		"backup.compaction.threshold",
+		"the maximum length of a backup chain before compaction occurs (0 to disable)",
+		0,
+		settings.WithVisibility(settings.Reserved),
+	)
+
+	backupCompactionWindow = settings.RegisterIntSetting(
+		settings.ApplicationLevel,
+		"backup.compaction.window",
+		"the number of backups to compact per compaction (must be greater than one and less than threshold)",
+		3,
+		settings.WithVisibility(settings.Reserved),
+	)
+
+	scheduleOnly = true // Test knob to only compact backups on a schedule.
```

Comment:
@jeffswenson I remember you expressing some level of aversion towards the "hookable global variable for testing" design, maybe a test knob would be better.

Reply:
+1 i would use a testing knob instead of a global hook